qcodes/instrument_drivers/rohde_schwarz/ZNB.py

```diff
@@ -1,16 +1,143 @@
+        assert isinstance(self.instrument, ZNBChannel)
+        i, q = self.instrument._get_cw_data(self._check_cw_sweep_first)
+        return i, q
+
+
+class FixedFrequencyPointIQ(MultiParameter):
+    """
+    Similar to the FixedFrequencyTraceIQ but now returns the mean of the
+    result, useful for two-tone and and other bigger sweeps where you do not
+    want to store all individual I-Q values.
+
+    Args:
+        name: parameter name
+        instrument: instrument the parameter belongs to
+        check_cw_sweep_first: flag to check if the vna is setup correctly
+    """
+
+    def __init__(self, name: str, instrument: "ZNBChannel",
+                 check_cw_sweep_first: bool = True) -> None:
+        super().__init__(name, names=("", ""), shapes=((), ()))
+        self._instrument = instrument
+        self._check_cw_sweep_first = check_cw_sweep_first
+        self.names = ('I','Q')
+        self.labels = (f'{instrument.short_name} I',
+                       f'{instrument.short_name} Q')
+        self.units = ('', '')
+        self.setpoints = ((), ()),
+
+    def get_raw(self) -> Tuple[float, float]:
+        """
+        Gets the mean of the raw real and imaginary part of the data. If the
+        flag `check_cw_sweep_first` is set to `True` then
+        at the cost of a few ms overhead checks if the vna is setup correctly.
+        """
+        assert isinstance(self.instrument, ZNBChannel)
+        i, q = self.instrument._get_cw_data(self._check_cw_sweep_first)
+        return np.mean(i), np.mean(q)
+
+
+class FixedFrequencyPointMagPhase(MultiParameter):
+    """
+    Similar to the FixedFrequencyTraceIQ but now returns the magnitude of
+    the mean of the result and it's phase.
+
+    Args:
+        name: parameter name
+        instrument: instrument the parameter belongs to
+        check_cw_sweep_first: flag to check if the vna is setup correctly
+    """
+
+    def __init__(self, name: str, instrument: "ZNBChannel",
+                 check_cw_sweep_first: bool = True) -> None:
+        super().__init__(name, names=("", ""), shapes=((), ()))
+        self._instrument = instrument
+        self._check_cw_sweep_first = check_cw_sweep_first
+        self.names = ('magnitude', 'phase')
+        self.labels = (f'{instrument.short_name} magnitude',
+                       f'{instrument.short_name} phase')
+        self.units = ('', 'rad')
+        self.setpoints = ((), ()),
+
+    def get_raw(self) -> Tuple[float, ...]:
+        """
+        Gets the magnitude and phase of the mean of the raw real and imaginary
+        part of the data. If the flag `check_cw_sweep_first` is set to `True`
+        for the instrument then at the cost of a few ms overhead checks if
+        the vna is setup correctly.
+        """
+        assert isinstance(self.instrument, ZNBChannel)
+        i, q = self.instrument._get_cw_data(self._check_cw_sweep_first)
+        s = np.mean(i) + 1j*np.mean(q)
+        return np.abs(s), np.angle(s)
+
```

Comment:
Yes, I think so too. FixedFrequencyPointMagPhase and FixedFrequencyPointIQ can inherit from FixedFrequencyTraceIQ.

Reply:
see the conversation above, it could in principle inherit from the FixedFrequencyTraceIQ, but actually most of the difference is contained in the setpoints set in the init. All shared functionality is already in the top instrument _get_cw_data function anyway so it think this is fine to keep it

Reply:
perfecto!